src/engraving/rendering/dev/systemlayout.cpp

```diff
@@ -393,6 +393,19 @@ System* SystemLayout::collectSystem(LayoutContext& ctx)
                     m->stretchToTargetWidth(oldWidth);
                     MeasureLayout::layoutMeasureElements(m, ctx);
                     BeamLayout::restoreBeams(m, ctx);
+                    for (size_t track = 0; track < ctx.dom().ntracks(); ++track) {
+                        Segment* s = m->findFirstR(SegmentType::EndBarLine, Fraction(1, 1));
```

Comment:
Some thoughts/questions:
- Fraction(1,1) is the length of a whole note, i.e. 4 quarters. So, it seems that this will only find the EndBarline segment if the measure has a length of 4 quarters or less (e.g. when the time signature is 4/4), and otherwise not (e.g. with a 5/4 time signature). 
- Doesn't this duplicate the calls to `layoutBarLine2` in `PageLayout::collectPage`?
- It looks like `layoutBarLine2` depends on the distances between staves (namely in `BarLine::calcY`), but as far as I know, that information is not yet known in `SystemLayout::collectSystem`. So I wonder why it still works :)